golang/context_test.go

```diff
@@ -0,0 +1,35 @@
+package tchannel
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type mockIncomingCall struct {
+	callerName string
+}
+
+func (m *mockIncomingCall) CallerName() string {
+	return m.callerName
+}
+
+var (
+	cn = "hello"
+)
+
+func TestWrapContextForTest(t *testing.T) {
+	call := &mockIncomingCall{callerName: cn}
+	ctx, cancel := NewContext(time.Second)
+	defer cancel()
+	actual := WrapContextForTest(ctx, call)
+	assert.Equal(t, call, actual.Value(contextKeyCall), "Incorrect call object returned.")
```

Comment:
instead of using the .Value, can you use the CurrentCall method to make sure they're the same